### ISSUE TITLE
Return provider_name, provider_display, and provider_display?

### ISSUE THREAD

**Issue by pdehaan** [mozilla/page-metadata-service#90]
We currently get these from Embed.ly, and we _may_ be using them in Activity Stream.
Notably in this mockup, we want to display "fivethirtyeight" next to a highlight: https://github.com/mozilla/activity-stream/pull/1153#issuecomment-243483512

Currently Embedly returns a `provider_name`, `provider_display`, and `provider_url`.
I think we partially had these before but they were removed because they were redundant at the time: https://github.com/mozilla/page-metadata-service/issues/27

Given the following URL, we'd probably want to do the following magic:

http://www.fivethirtyeight.com/features/the-broncos-new-quarterback-is-inexperienced-but-at-least-hes-not-peyton-manning/

- `provider_url`: http://fivethirtyeight.com &mdash; just the base protocol, and domain of the canonical URL, but no path.

- `provider_display`: Looks like just the `provider_url`, minus the protocol. 

- `provider_name`: FiveThirtyEight  &mdash; possibly extracted from the `og:site_name` or `twitter:site` metadata (minus leading "@"). If not available maybe just the `provider_url`, minus the protocol, "www.", and TLD (which would mean we'd have to know the difference between dual TLDs like bbc.co.uk versus www.bbc.com).

http://embed.ly/docs/explore/extract?url=http%3A%2F%2Fwww.fivethirtyeight.com%2Ffeatures%2Fthe-broncos-new-quarterback-is-inexperienced-but-at-least-hes-not-peyton-manning%2F

- `provider_url`: http://fivethirtyeight.com
- `provider_name`: FiveThirtyEight
- `provider_display`: fivethirtyeight.com

**Comment by pdehaan** [comment]
A slightly verbose way to do it using [**tldjs**](http://npm.im/tldjs) and `url.parse()`:

```js
var tld = require('tldjs');
const url = require('url');

const domain = "https://www.bbc.co.uk/foo/bar.html";

const parsed = url.parse(domain);
// Inject the values from tldjs:
parsed.tldExists = tld.tldExists(domain); // true
parsed.domain = tld.getDomain(domain); // bbc.co.uk
parsed.subdomain = tld.getSubdomain(domain); // www
parsed.publicSuffix = tld.getPublicSuffix(domain); // co.uk
parsed.isValid = tld.isValid(domain); // true

console.log(JSON.stringify(parsed, null, 2));
```

<details>
<summary>**OUTPUT:**</summary>
```json
{
  "protocol": "https:",
  "slashes": true,
  "auth": null,
  "host": "www.bbc.co.uk",
  "port": null,
  "hostname": "www.bbc.co.uk",
  "hash": null,
  "search": null,
  "query": null,
  "pathname": "/foo/bar.html",
  "path": "/foo/bar.html",
  "href": "https://www.bbc.co.uk/foo/bar.html",
  "tldExists": true,
  "domain": "bbc.co.uk",
  "subdomain": "www",
  "publicSuffix": "co.uk",
  "isValid": true
}
```
</details>

**Comment by jaredkerim** [comment]
Fixed by #120
